src/components/ec-datepicker/ec-datepicker.spec.js

```diff
@@ -0,0 +1,387 @@
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-23').attributes('title')).toBe(undefined);
+
+      expect(calendarWrapper.element).toMatchSnapshot('calendar');
+    });
+
+    it('should have disabled weekends', () => {
+      const { calendarWrapper } = mountDatepicker({
+        areWeekendsDisabled: true,
+      });
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-19').classes('flatpickr-disabled')).toBeTruthy();
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-20').classes('flatpickr-disabled')).toBeTruthy();
+
+      expect(calendarWrapper.element).toMatchSnapshot('calendar');
+    });
+  });
+
+  it('should open the calendar when we click on the input icon', () => {
+    const { inputWrapper, calendarWrapper } = mountDatepicker();
+
+    inputWrapper
+      .findByDataTest('ec-input-field__icon-wrapper')
+      .trigger('click');
+
+    expect(calendarWrapper.element).toMatchSnapshot();
+  });
+
+  describe('@events', () => {
+    it('@ready - should be emitted when the calendar is in a ready state', () => {
+      const { inputWrapper } = mountDatepicker();
+
+      expect(inputWrapper.emitted('ready').length).toBe(1);
+    });
+
+    it('@open - should be emitted when the calendar opens', () => {
+      const { inputWrapper } = mountDatepicker();
+
+      inputWrapper
+        .findByDataTest('ec-datepicker')
+        .trigger('click');
+
+      expect(inputWrapper.emitted('open').length).toBe(1);
+    });
+
+    it('@close - should be emitted when the calendar closes', () => {
+      const { inputWrapper, calendarWrapper } = mountDatepicker({ value: '2022-02-20' });
+
+      inputWrapper
+        .findByDataTest('ec-datepicker')
+        .trigger('click');
+
+      calendarWrapper
+        .findByDataTest('ec-datepicker__calendar-day--2022-02-24')
+        .trigger('click');
+
+      expect(inputWrapper.emitted('close').length).toBe(1);
+    });
+  });
+
+  describe('watchers', () => {
+    it('should update options after initialization', async () => {
+      const { inputWrapper, calendarWrapper } = mountDatepicker({
+        options: {},
+      });
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-21').classes('flatpickr-disabled')).toBeFalsy();
+
+      await inputWrapper.setProps({
+        options: {
+          minDate: '2022-02-22',
+        },
+      });
+
+      inputWrapper.vm.$nextTick();
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-21').classes('flatpickr-disabled')).toBeTruthy();
+    });
+
+    it('should not update with custom hooks after initialization', async () => {
+      const { inputWrapper, calendarWrapper } = mountDatepicker({
+        options: {},
+      });
+
+      await inputWrapper.setProps({
+        options: {
+          onChange: () => {
+            inputWrapper.vm.$emit('changed');
```

Comment:
We shouldn't be using vm for this kind of tests. You can set `let onChangeSpy = jest.fn()`, pass it to `onChange: onChangeSpy` and then assert that the spy didn't get called.

Reply:
Hey thank you for feedback, can you please explain why is bad to use vm for this tests for future reference? Thank you in advance :)